### DISCUSSION TITLE
Custom HTTP header for trace ID

### DISCUSSION THREAD

**Discussion by nestorsokil** [open-telemetry/opentelemetry-java-instrumentation#2429] [Q&A]
Hey, I was wondering how I can supply my own propagator implementation to the agent? 
What I want to do is to accept trace ID from a custom HTTP header and then propagate to downstream services using the B3 standard.

Or is the custom propagator a wrong solution for this? I basically just need to use a custom header to seed the trace and then use the regular b3multi.

**Comment by iNikem** [comment]
Custom propagator seems like the correct solution. Unfortunately currently there is no way to supply one to the agent. Can you please open an issue to add such extension point?
